dali/pipeline/executor/executor.cc

```diff
@@ -238,9 +238,7 @@ void Executor::RunGPU() {
   previous_gpu_queue_idx_ = queue_idx;
 }
 
-void Executor::Outputs(DeviceWorkspace *ws) {
-  DALI_ENFORCE(ws != nullptr, "Workspace is nullptr");
-  ws->Clear();
+void Executor::ReleaseOutputs() {
```

Comment:
This is an API break - previously you could do runcpu/rungpu/outputs, now you need to do runcpu/rungpu/outputs/releaseoutputs.
At the very least you need to update all tests/benchmarks that will hang with this change. Is it not possible to have ReleaseOutputs as separate but also enable the old behavior?

Reply:
Added ShareOutputs and and implemented Outputs as call to ReleaseOutputs and ShareOutputs so old code should still work.